scripts/rules.go

```diff
@@ -63,6 +66,40 @@ func testingWithOwnerUser(m dsl.Matcher) {
 			m.File().Name.Matches(`_test\.go$`)).
 		At(m["SetupConfig"]).
 		Report(`The CLI will be operating as the owner user, which has unrestricted permissions. Consider creating a different user.`)
+
+	m.Match(`
+	$client, $_ := coderdenttest.New($t, $*_)
+	$*_
+	clitest.$SetupConfig($t, $client, $_)
+	`).Where(m["t"].Type.Implements("testing.TB") &&
+		m["SetupConfig"].Text.Matches("^SetupConfig$") &&
+		m.File().Name.Matches(`_test\.go$`)).
+		At(m["SetupConfig"]).
+		Report(`The CLI will be operating as the owner user, which has unrestricted permissions. Consider creating a different user.`)
+
+	// For the enterprise code, we check for any method called on the client.
+	// While we want to be a bit stricter here, some methods are known to require
+	// the owner user, so we exclude them.
+	m.Match(`
+		$client, $_ := coderdenttest.New($t, $*_)
+		$*_
+		$_, $_ := $client.$Method($*_)
```

Comment:
Nit: mis-matched indentation. Personally I'd prefer for everything to be indented like this, helps my brain parse it since it's deeper than the `m.Match`, but I'll settle for consistency. 😂

Reply:
Now that I see this, I can't un-see it. I prefer the indented too :D